code/datums/components/beacon.dm

```diff
@@ -126,17 +117,23 @@
 	playsound(source, 'sound/machines/twobeep.ogg', 15, 1)
 	user.visible_message("[user] activates [source]'s signal.")
 	user.show_message(span_notice("The [source] beeps and states, \"Your current coordinates were registered by the supply console. LONGITUDE [location.x]. LATITUDE [location.y]. Area ID: [get_area(source)]\""), EMOTE_AUDIBLE, span_notice("The [source] vibrates but you can not hear it!"))
-	beacon_datum = new /datum/supply_beacon("[user.name] + [A]", get_turf(source), user.faction)
+	beacon_datum = new /datum/supply_beacon("[user.name] + [Area]", get_turf(source), user.faction)
 	RegisterSignal(beacon_datum, COMSIG_QDELETING, PROC_REF(clean_beacon_datum))
+	RegisterSignal(source, COMSIG_MOVABLE_MOVED, PROC_REF(updatepos))
+	if(ismob(source.loc))
+		RegisterSignal(source.loc, COMSIG_MOVABLE_MOVED, PROC_REF(updatepos))
+	RegisterSignal(source, COMSIG_ITEM_EQUIPPED, PROC_REF(equip))
+	RegisterSignal(source, COMSIG_ITEM_DROPPED, PROC_REF(dropped))
 	SEND_GLOBAL_SIGNAL(COMSIG_GLOB_SUPPLY_BEACON_CREATED, src)
 	source.update_appearance()
 
 ///Deactivates the beacon
 /datum/component/beacon/proc/deactivate(atom/movable/source, mob/user)
-	if(length(user?.do_actions))
-		user.balloon_alert(user, "Busy!")
-		active = TRUE
-		return
+	if(user)
```

Comment:
Okay like... why did you do this? You're just adding this user check for no reason